cmd/snap-recovery/partition/sfdisk.go

```diff
@@ -173,12 +173,13 @@ func deviceName(name string, index int) string {
 	return fmt.Sprintf("%s%d", name, index)
 }
 
-// buildPartitionList builds a list of partitions based on the current device contents
-// and gadget structure list, in sfdisk dump format. Return a partitioning description
-// suitable for sfdisk input and a role to device node map.
-func buildPartitionList(ptable *sfdiskPartitionTable, pv *gadget.LaidOutVolume) (*bytes.Buffer, map[string]string) {
+// buildPartitionList builds a list of partitions based on the current
+// device contents and gadget structure list, in sfdisk dump
+// format. Return a partitioning description suitable for sfdisk input
+// and a list of the partitions to be created
+func buildPartitionList(ptable *sfdiskPartitionTable, pv *gadget.LaidOutVolume) (sfdiskInput *bytes.Buffer, toBeCreated map[string]gadget.LaidOutStructure) {
 	buf := &bytes.Buffer{}
-	deviceMap := map[string]string{}
+	toBeCreated = make(map[string]gadget.LaidOutStructure, len(pv.LaidOutStructure))
```

Comment:
Probably a followup material, but we could do:
```
type deviceStructure {
    gadget.LaidOutStructure
    Node string
}
```

Reply:
Good suggestion, I implemented this now as []deviceStructure sorted by layout as defined in the gadget.yaml